cpp/src/lance/io/reader.cc

```diff
@@ -156,19 +151,14 @@ ::arrow::Result<::std::shared_ptr<::arrow::Scalar>> FileReader::GetListScalar(
   decoder->Reset(pos, length);
   ARROW_ASSIGN_OR_RAISE(auto offsets_arr, decoder->ToArray(idx, 2));
   auto offsets = std::static_pointer_cast<::arrow::Int32Array>(offsets_arr);
+  if (offsets->Value(0) == offsets->Value(1)) {
```

Comment:
so we're not using the null bitmap to check?

Reply:
Null bitmap is not actually stored physically. It only exists during in-memory reconstruction.